src/main/java/com/spring_boot/repository/MailDao.java

```diff
@@ -0,0 +1,95 @@
+package com.spring_boot.repository;
+
+import com.home_work.spring_boot.entity.Letter;
+import com.home_work.spring_boot.services.PropertiesService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.io.*;
+import java.util.LinkedList;
+import java.util.Map;
+
+@Repository
+public class MailDao {
+
+    private static final String MAILS_TO_SEND = "fileNameToSend";
+    private static final String SENT_MAILS = "fileNameOfSent";
+
+
+    private PropertiesService propertiesService;
+
+    @Autowired
+    public MailDao(PropertiesService propertiesService) {
+        this.propertiesService = propertiesService;
+    }
+
+    public void saveMail(Letter letter, String fileToSave) {
+        LinkedList<Letter> letters = getMails(fileToSave);
+        giveId(letter, letters);
+        letters.add(letter);
+        outPutStream(letters, fileToSave);
+    }
+
+    private void outPutStream(LinkedList<Letter> letters, String fileToSave) {
+        try (OutputStream outputStream = new FileOutputStream(getPathAndNameOfFile(fileToSave));
+             ObjectOutputStream objectOutputStream = new ObjectOutputStream(outputStream)) {
+
+            objectOutputStream.writeObject(letters);
+            objectOutputStream.flush();
+
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
It's poor handling of an exception. In this case, this error is unrecoverable and application should stop: you should propagate this error at the top, instead of masking it.